test/test_svi.py

```diff
@@ -67,3 +68,26 @@ def body_fn(i, val):
     opt_state, rng = fori_loop(0, 300, body_fn, (opt_state, rng_train))
     params = get_params(opt_state, rng, guide_args=())
     assert_allclose(params['loc'], true_coef, atol=0.05)
+
+
+def test_dynamic_supports():
+    x_prior = dist.TransformedDistribution(
+        dist.Normal(), [AffineTransform(0, 2), SigmoidTransform(), AffineTransform(0, 3)])
+    x_guide = dist.Uniform(0, 3)
+
+    def model():
+        sample('x', x_prior)
+
+    def guide():
+        sample('x', x_guide)
+
+    opt_init, opt_update, get_opt_params = optimizers.adam(0.01)
+    # set base value of x_guide is 10
+    x_base = 10
+    guide = substitute(guide, base_param_map={'x': x_base})
+    svi_init, _, svi_eval = svi(model, guide, elbo, opt_init, opt_update, get_opt_params)
+    opt_state, get_params = svi_init(random.PRNGKey(0), (), ())
+    actual_loss = svi_eval(random.PRNGKey(1), opt_state)
```

Comment:
Hmm..this seems to be giving nan values.

Reply:
...

Reply:
My bad, 10 is out of support of Uniform(0, 1).

Reply:
I changed `10` to `0.9` and added an assert statement that `actual_loss` is finite.